packages/x-charts/src/SparkLineChart/SparkLineChart.tsx

```diff
@@ -159,7 +159,10 @@ const SparkLineChart = React.forwardRef(function SparkLineChart(props: SparkLine
         axisHighlight?.y === 'none'
       }
     >
-      {plotType === 'bar' && <BarPlot slots={slots} slotProps={slotProps} />}
+      {plotType === 'bar' && (
+        <BarPlot slots={slots} slotProps={slotProps} sx={{ shapeRendering: 'auto' }} />
```

Comment:
Would it make sense to change the `shapeRendering` for all bar elements?

https://github.com/mui/mui-x/blob/404c2bb2f268d4e1713ebebe78bb37a48f48a163/packages/x-charts/src/BarChart/BarElement.tsx#L55

Reply:
It has been introduced for when bar are close to each other or stack. Without it border between two rects can be a bit messy due to (I assume) anti-aliasing.

So I remote it in sparkline because this should never get multiple series/stacked bars (I hope). But for main bar chart I prefer to keep it

https://github.com/mui/mui-x/assets/45398769/686a5dec-97d8-48e9-a70a-8f350a1340a0

Reply:
Gotcha!